app/ts/store/notice.ts

```diff
@@ -0,0 +1,18 @@
+import { signal } from '@preact/signals'
+import { Optional } from '../types.js'
+
+export type Notice = {
+	id: number
+	title: string
+	message: string
+}
+
+const notices = signal<Notice[]>([])
+
+export function useNotice() {
+	const notify = (notice: Optional<Notice, 'id'>) => {
```

Comment:
It feels like `notify` should not allow caller provided `id` because it is very possible/likely that two callers will use the same ID.  by having this function manage IDs we don't have to worry about collisions.

On a related note, instead of `notices.value.length + 1` we should just have a file-local `let id = 0` and use `++id` for the ID of each notice.  This way they never get reused (which could cause some problems).

Reply:
removed option for notice id and signal computed the id instead